### PR TITLE
feat: add auto-merging approved PRs for GitHub Actions

### PR DESCRIPTION
**Pull request by dr460nf1r3**
This is a pre-requisite to make reviewing PRs and auto-deploying PRs by non-chaotic-aur members feasible. I'm picturing a separate org group with no write access, who could still review the stuff without causing additional actions needed by the actual staff.

A PAT is *required* to trigger the on-commit event. Using GITHUB_TOKEN, this does not get triggered and fails any scheduled pipeline after the merge event.

### CODE REVIEW

File: .github/workflows/automerge.yml
Comment (JustTNE) [inline .github/workflows/automerge.yml:20]: I doubt this is very secure

**Review by JustTNE — CHANGES_REQUESTED** [review]
All in all, this seems to let anyone merge PRs by approving them, is that wrong? 
And it seems to use a very easy to fake parameter to decide if it should be merged or not.

File: .github/workflows/automerge.yml
Comment (dr460nf1r3) [reply .github/workflows/automerge.yml:20]: This was made under the assumption that a branch rule requires at least 1 approval of code owners.

**Pull request closed without merging**
